atomic_reactor/utils/imageutil.py

```diff
@@ -5,18 +5,95 @@
-    return {}
+from osbs.utils import ImageName
+
+from atomic_reactor import config
+from atomic_reactor import util
+from atomic_reactor.types import ImageInspectionData
+
+
+def inspectable(image: Union[str, ImageName]) -> bool:
+    """Check if we should expect the image to be inspectable."""
+    im = str(image)
+    return not (util.base_image_is_scratch(im) or util.base_image_is_custom(im))
+
+
+class ImageUtil:
+    """Convenience class for working with images relevant to the build process.
+
+    Supports e.g. inspecting the base image and other parent images.
+    """
+
+    def __init__(self, dockerfile_images: util.DockerfileImages, conf: config.Configuration):
+        """Init an ImageUtil.
+
+        :param dockerfile_images: information about the image references in the Dockerfile
+        :param conf: atomic-reactor configuration
+        """
+        self._dockerfile_images = dockerfile_images
+        self._conf = conf
+
+    def set_dockerfile_images(self, dockerfile_images: util.DockerfileImages) -> None:
+        """Set a new dockerfile_images instance."""
+        self._dockerfile_images = dockerfile_images
+
+    def get_inspect_for_image(
+        self, image: Union[str, ImageName], platform: Optional[str] = None
+    ) -> ImageInspectionData:
+        """Inspect an image. Should mainly be used to inspect parent images.
+
+        The image must be inspectable, passing a non-inspectable image is an error.
+
+        The result is cached, this method will not query the registry more than once
+        (if successful) for the same image reference (within one Python process).
+
+        :param image: The image to inspect
+        :param platform: Optionally, inspect the base image for a specific platform.
+            This can be either the platform name (e.g. x86_64) or the GOARCH name (amd64).
+        """
+        if not inspectable(image):
+            raise ValueError(f"{image!r} is not inspectable")
+
+        goarch = self._conf.platform_to_goarch_mapping[platform]
+        return self._cached_inspect_image(str(image), goarch)
+
+    def base_image_inspect(self, platform: Optional[str] = None) -> ImageInspectionData:
+        """Inspect the base image (the parent image for the final build stage).
+
+        If the base image is scratch or custom, return an empty dict.
+
+        :param platform: Optionally, inspect the base image for a specific platform.
+            This can be either the platform name (e.g. x86_64) or the GOARCH name (amd64).
+        """
+        base_image: Union[str, ImageName] = self._dockerfile_images.base_image
+        if not inspectable(base_image):
+            return {}
+
+        return self.get_inspect_for_image(base_image, platform)
+
+    @functools.lru_cache(maxsize=None)
+    def _cached_inspect_image(
+        self, image: str, goarch: Optional[str] = None
+    ) -> ImageInspectionData:
+        # Important: this method must take the image name as a string, not an ImageName.
```

Comment:
and why don't you just convert ImageName (if provided) to string and then parse it again ?

Reply:
but well if it will be used just internally here it doesn't matter

Reply:
> and why don't you just convert ImageName (if provided) to string and then parse it again ?

Yeah that's what I'm doing. `get_inspect_for_image` accepts both string and ImageName, converts to string, and `_cached_inspect_image`  parses back to ImageName.